streams/src/main/java/org/apache/kafka/streams/kstream/KWindowedTable.java

```diff
@@ -0,0 +1,111 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream;
+
+/**
+ * KWindowedTable is an abstraction of a change log stream on a window basis.
+ *
+ *
+ * @param <K> the type of keys
+ * @param <V> the type of values
+ * @param <W> the type of window definition
+ */
+public interface KWindowedTable<K, V, W extends Window> {
```

Comment:
KWindowedTable is a bit awkward. What if we just called it WindowedTable, or, if we have to be consistent, WindowedKTable?